### PR TITLE
Add asset compilation and collection before deploy

### LINKED ISSUE #120: Compile and collect USWDS assets on staging
Now that we use USWDS on our pages, we need to compile and "deploy" those assets in our Github Actions workflow that deploys from `main` to the staging environment. The assets are compiled and deployed with `npm` so we need to `npm install`, `npx gulp compile` and `npx gulp copyAssets`. 

So that we don't have to install Node in our Github Actions host, we can use our existing docker-compose setup to run these commands. Probably `docker compose run node npm install`, `docker compose run node npx gulp compile` and `docker compose run node npx gulp copyAssets` at a first guess, if we got the docker compose setup right?

### PR DESCRIPTION
**Pull request by igorkorenfeld**
## 🗣 Description ##

This PR adds two steps to the deploy GitHub Action that compile the USWDS assets and collect them into the `/public` folder using our docker container before deploying them to staging. By using the GH action and the docker container we can avoid having to run `collectstatic` on the server upon deploying. 

This closes #120 

https://getgov-staging.app.cloud.gov/

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ on:
   push:
     branches:
       - main
+      - ik/stable-uswds-assets
```
Comment (SSPJ) [inline .github/workflows/deploy.yaml:12]: Remove before merging 🙂

**Review by SSPJ — APPROVED** [review]
LGTM 👍 Nice work, Igor. 🎉 

- [x] I checked that this action ran successfully

**Comment by neilmb** [comment]
> @igorkorenfeld is there anything similar to this one will need to do to deploy to unstable locally? Is all of that already documented? https://github.com/cisagov/getgov/tree/main/docs/operations#deploy

I'm fine with this being merged as-is, but would love to see a follow-on ticket or PR that would make an `ops/scripts/build_and_deploy.sh` that would (at this point) run the asset compilation, the collectstatic, and the `cf push`. We could use that script as developers to deploy to unstable and also use it in the Github action to deploy to staging.

File: docs/operations/README.md [hunk 2]
```diff
@@ -36,9 +36,15 @@ Binding the database in `manifest-<ENVIRONMENT>.json` automatically inserts the
 
 # Deploy
 
-We have two environments: `unstable` and `staging`. Developers can deploy locally to unstable whenever they want. However, only our CD service can deploy to `staging`, and it does so on every commit to `main`. This is to ensure that we have a "golden" environment to point to, and can still test things out in an unstable space. To deploy locally to `unstable`:
+We have two environments: `unstable` and `staging`. Developers can deploy locally to unstable whenever they want. However, only our CD service can deploy to `staging`, and it does so on every commit to `main`. This is to ensure that we have a "golden" environment to point to, and can still test things out in an unstable space. You should make sure all of the USWDS assets are compiled and collected before deploying to unstable. To deploy locally to `unstable`:
 
 ```bash
+# Compile and collect the assets
```
Comment (neilmb) [inline docs/operations/README.md:42]: These instructions are great!

**Review by neilmb — APPROVED** [review]
So great to see this working on staging. Thanks Igor!

File: docs/operations/README.md (hunk 2 above)
Comment (igorkorenfeld) [reply docs/operations/README.md:42]: Thank you!

**Comment by igorkorenfeld** [comment]
@neilmb that sounds good, I can do some follow-on work

**Pull request merged**